server/src/main/java/keywhiz/service/resources/BatchSecretDeliveryResource.java

```diff
@@ -0,0 +1,156 @@
+ * limitations under the License.
+ */
+
+package keywhiz.service.resources;
+
+import com.codahale.metrics.annotation.ExceptionMetered;
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.auth.Auth;
+import keywhiz.api.BatchSecretRequest;
+import keywhiz.api.SecretDeliveryResponse;
+import keywhiz.api.model.Client;
+import keywhiz.api.model.SanitizedSecret;
+import keywhiz.api.model.Secret;
+import keywhiz.service.config.Readonly;
+import keywhiz.service.daos.AclDAO;
+import keywhiz.service.daos.AclDAO.AclDAOFactory;
+import keywhiz.service.daos.ClientDAO;
+import keywhiz.service.daos.ClientDAO.ClientDAOFactory;
+import keywhiz.service.daos.SecretController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import java.util.Iterator;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+/**
+ * parentEndpointName secret
+ * <p>
+ * resourceDescription Retrieve Batched Named Secrets
+ */
+@Path("/batchsecret")
+@Produces(APPLICATION_JSON)
+public class BatchSecretDeliveryResource {
+  private static final Logger logger = LoggerFactory.getLogger(BatchSecretDeliveryResource.class);
+
+  private final SecretController secretController;
+  private final AclDAO aclDAO;
+  private final ClientDAO clientDAO;
+
+  @Inject
+  public BatchSecretDeliveryResource(@Readonly SecretController secretController,
+                                     AclDAOFactory aclDAOFactory, ClientDAOFactory clientDAOFactory) {
+    this.secretController = secretController;
+    this.aclDAO = aclDAOFactory.readonly();
+    this.clientDAO = clientDAOFactory.readwrite();
+  }
+
+  @VisibleForTesting
+  BatchSecretDeliveryResource(SecretController secretController, AclDAO aclDAO,
+                              ClientDAO clientDAO) {
+    this.secretController = secretController;
+    this.aclDAO = aclDAO;
+    this.clientDAO = clientDAO;
+  }
+
+  /**
+   * Retrieve Secret by name
+   *
+   * @param secrets the name of the Secrets to retrieve in batch
+   * @param client  the client performing the retrieval
+   * @return the secret with the specified name, if present and accessible to the client
+   * <p>
+   * responseMessage 200 Found and retrieved Secret with given name
+   * responseMessage 403 Secret is not assigned to Client
+   * responseMessage 404 Secret with given name not found
+   * responseMessage 500 Secret response could not be generated for given Secret
+   */
+  @Timed
+  @ExceptionMetered
+  @POST
+  @Consumes(APPLICATION_JSON)
+  public List<SecretDeliveryResponse> getBatchSecret(@Auth Client client, @Valid BatchSecretRequest request) {
+
+
+    List<SanitizedSecret> sanitizedSecrets = aclDAO.getBatchSanitizedSecretsFor(client, request.secrets);
```

Comment:
I'd rename these (and the booleans below) to something like `clientAccessibleSecrets` and `existingSecrets` to make the significance of a secret missing from one or the other list clearer.